src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1533,6 +1534,21 @@ InferenceProfiler::DetermineStatsModelVersion(
   return cb::Error::Success;
 }
 
+// Only for unit-testing
+#ifndef DOCTEST_CONFIG_DISABLE
+cb::Error
+InferenceProfiler::SetTopLevelResponseCaching(
+    bool enable_top_level_response_caching)
+{
+  parser_ = std::make_shared<ModelParser>(cb::BackendKind::TRITON);
+  if (parser_ == nullptr) {
+    return cb::Error("Null Pointer Exception");
```

Comment:
This error message could be more descriptive. Something about failing to initialize the ModelParser perhaps